--- a/gcc/cp/decl.c
+++ b/gcc/cp/decl.c
@@ -7173,6 +7173,18 @@
   return decl;
 }
 
+#undef builtin_function
+
+tree
+builtin_function (name, type, function_code, library_name)
+     const char *name;
+     tree type;
+     enum built_in_function function_code;
+     const char *library_name;
+{
+  return define_function (name, type, function_code, (void (*) PROTO((tree))) pushdecl, library_name);
+}
+
 /* When we call finish_struct for an anonymous union, we create
    default copy constructors and such.  But, an anonymous union
    shouldn't have such things; this function undoes the damage to the
