--- a/gcc/cp/decl.c
+++ b/gcc/cp/decl.c
@@ -7189,6 +7189,13 @@
 
 #undef builtin_function
 
+static void
+pushdecl_1 (decl)
+     tree decl;
+{
+  pushdecl (decl);
+}
+
 tree
 builtin_function (name, type, function_code, library_name)
      const char *name;
@@ -7196,7 +7203,8 @@
      enum built_in_function function_code;
      const char *library_name;
 {
-  return define_function (name, type, function_code, (void (*) PROTO((tree))) pushdecl, library_name);
+  return define_function (name, type, function_code, pushdecl_1,
+                          library_name);
 }
 
 /* When we call finish_struct for an anonymous union, we create
